src/app/beer_garden/garden.py

```diff
@@ -133,3 +116,29 @@ def garden_add_system(system: System, garden_name: str):
 @publish_event(Events.GARDEN_UPDATED)
 def update_garden(garden: Garden):
     return db.update(garden)
+
+
+def handle_event(event):
+    """Handle garden-related events
+
+    For GARDEN events we only care about events originating from downstream. We also
+    only care about immediate children, not grandchildren.
+
+    Whenever a garden event is detected we should update that garden's database
+    representation.
+
+    This method should NOT update the routing module. Let its handler worry about that!
+    """
```

Comment:
This is an interesting way to handle the events. So anything that would impact the DB layer from downstream events would be processed here.

Reply:
Yeah, things got to the point where I wasn't able to reason about what was going on the event handling side of things. So I split things out so that each module has its own event handler function that will handle the events that it (and only it) cares about.